app/models/service_interaction.rb

```diff
@@ -1,4 +1,12 @@
 class ServiceInteraction < ActiveRecord::Base
   validates :service_identifier, :service_label, :interaction_identifier,
   :interaction_label, presence: true, uniqueness: true
+
+  def self.find_service(identifier)
+    Service.find_by identifier: identifier
+  end
+
+  def self.find_interaction(identifier)
+    Interaction.find_by identifier: identifier
+  end
```

Comment:
Probably my fault for not being clearer when I suggested it, but what I expected here is that we would have methods on the instance rather than the class.  E.g. we could do:

```ruby
service_interaction = ServiceInteraction.find(...)

service_interaction.service # returns the Service instance with the same identifier as service_instance.service_identifier

service_interaction.interaction # returns the Interaction instance with the same identifier as service_instance.interaction_identifier
```

This gives us an API similar to what we'd get if we used ActiveRecord belongs_to / has_one style associations (and leaves the door open to us doing that in the future if we think it makes sense).  The implementation would be almost identical, and the tests too, just not defined as `def self.find_blah` and not taking an argument.